File: Java/Lab1/BankAccount.java

public class BankAccount {
	private long ID;
	private double balance;
	public BankAccount(long ID, double balance)
	{
		this.ID= ID;
		this.balance= balance;
	}
	public void deposit(double value)
	{
		if (value> 0)
			balance= balance+ value;
	}
	public void withdraw(double value)
	{
		if (value >0 && balance>= value)
			balance= balance- value;
	}
	public void ShowDetails()
	{
		System.out.println("ID: "+ ID);
		System.out.println("Balance: "+ balance);
	}
public static void main(String args[])
{
	BankAccount acc1= new BankAccount(123456789, 30000000);
	BankAccount acc2= new BankAccount(123456788, 45000000);
	
	acc1.ShowDetails();
	acc1.deposit(2000000);
	acc1.ShowDetails();
	
	acc2.ShowDetails();
	acc2.withdraw(100000);
	acc2.ShowDetails();
}
}

File: bankAccount.cpp
#include <iostream>
#include <conio.h>

using namespace std;

class BankAccount
{
    private:    long id;
                double balance;
    public:
        BankAccount(long id, double balance)
        {
            this->id = id;
            this->balance = balance;
        }
        void deposit(double dps)
        {
            if (dps> 0)
                balance= balance + dps;
        }
        void withdraw(double wdr)
        {
            if(wdr> 0 && balance > wdr)
                balance= balance- wdr;
            else cout<<"Invalid Withdrawal"<<endl;
        }
        void showDetails()
        {
            cout << "ID= " << id << endl;
            cout << "Balance= " << balance << endl;
            cout << "\n\n";
        }
};

int main()
{
    BankAccount acc1(123456789, 500000);
    BankAccount acc2(123456788,800000);

    acc1.showDetails();
    acc1.deposit(20000);
    acc1.showDetails();

    acc2.showDetails();
    acc2.withdraw(40000);
    acc2.showDetails();

    return 0;
}
